src/import_edf.jl

```diff
@@ -147,23 +144,21 @@ end
 #####
 
 function extract_channels(edf_signals, channel_matchers)
-    extracted_channel_names = String[]
-    extracted_channels = EDF.Signal[]
+    extracted_channels = Pair{String,EDF.Signal}[]
     for channel_matcher in channel_matchers
         for edf_signal in edf_signals
             edf_signal isa EDF.Signal || continue
-            any(x -> x === edf_signal, extracted_channels) && continue
+            any(x -> x === edf_signal, map(last, extracted_channels)) && continue
             channel_name = channel_matcher(edf_signal)
             channel_name === nothing && continue
-            push!(extracted_channel_names, channel_name)
-            push!(extracted_channels, edf_signal)
+            push!(extracted_channels, channel_name => edf_signal)
         end
     end
-    return extracted_channel_names, extracted_channels
+    return extracted_channels
```

Comment:
We'll have to register this change as a breaking version, right, since the return type is changing? Or is this function internal only?

Reply:
I'd say `extract_channels_by_label` is the public-facing entrypoint, and `extract_channels` is internal.